WordPressAuthenticator/OAuth/ProofKeyForCodeExchange.swift

```diff
@@ -0,0 +1,41 @@
+// https://developers.google.com/identity/protocols/oauth2/native-app#step1-code-verifier
+//
+// FIXME: follow spec!
+//
+// A code_verifier is a high-entropy cryptographic random string using the unreserved
+// characters [A-Z] / [a-z] / [0-9] / "-" / "." / "_" / "~", with a minimum length of 43
+// characters and a maximum length of 128 characters.
+//
+// The code verifier should have enough entropy to make it impractical to guess the value.
+struct ProofKeyForCodeExchange {
+
+    enum Mode {
+        case s256
+        case plain
+
+        var method: String {
+            switch self {
+            case .plain: return "plain"
+            case .s256: return "S256"
+            }
+        }
+    }
+
+    let codeVerifier: String
+    let mode: Mode
+
+    init(codeVerifier: String, mode: Mode) {
```

Comment:
Considering we should randomly generate `codeVerifier`, do you think we can make this argument nil by default and generate a random string internally?

Reply:
Thank you for the suggestion. I'm keen on generating the string internally according to the spec but [I think that's work that could live in a dedicated PR](https://github.com/wordpress-mobile/WordPressAuthenticator-iOS/pull/717/files#r1043267507). Does that sound okay?

What I have in mind is along the lines of your suggestion, with the difference of providing the random string as the default value.

```swift
init(codeVerifier: String = makeRandomPKCEString(), method: Method = .s256) {
```